include/alpaka/core/UniformCudaHip.hpp

```diff
@@ -0,0 +1,133 @@
+                    ALPAKA_API_PREFIX(GetLastError)();
+                    throw std::runtime_error(sError);
+                }
+            }
+            //-----------------------------------------------------------------------------
+            //! CUDA/Hip runtime API error checking with log and exception, ignoring specific error values
+            // NOTE: All ignored errors have to be convertible to Error_t.
+            template<
+                typename... TErrors>
+            ALPAKA_FN_HOST auto rtCheckIgnore(
+                Error_t const & error,
+                char const * cmd,
+                char const * file,
+                int const & line,
+                TErrors && ... ignoredErrorCodes)
+            -> void
+            {
+                if(error != ALPAKA_API_PREFIX(Success))
+                {
+                    std::array<Error_t, sizeof...(ignoredErrorCodes)> const aIgnoredErrorCodes{ignoredErrorCodes...};
+
+                    // If the error code is not one of the ignored ones.
+                    if(std::find(aIgnoredErrorCodes.cbegin(), aIgnoredErrorCodes.cend(), error) == aIgnoredErrorCodes.cend())
+                    {
+                        rtCheck(error, ("'" + std::string(cmd) + "' returned error ").c_str(), file, line);
+                    }
+                }
+            }
+            //-----------------------------------------------------------------------------
+            //! CUDA runtime API last error checking with log and exception.
+            ALPAKA_FN_HOST inline auto rtCheckLastError(
+                char const * desc,
+                char const * file,
+                int const & line)
+            -> void
+            {
+                Error_t const error(ALPAKA_API_PREFIX(GetLastError)());
+                rtCheck(error, desc, file, line);
+            }
+        }
+    }
+}
+
+#if BOOST_COMP_MSVC
+    //-----------------------------------------------------------------------------
+    //! CUDA runtime error checking with log and exception, ignoring specific error values
+    #define ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE(cmd, ...)\
+        ::alpaka::uniform_cuda_hip::detail::rtCheckLastError("'" #cmd "' A previous CUDA call (not this one) set the error ", __FILE__, __LINE__);\
+        ::alpaka::uniform_cuda_hip::detail::rtCheckIgnore(cmd, #cmd, __FILE__, __LINE__, __VA_ARGS__)
+#else
+    #if BOOST_COMP_CLANG
+        #pragma clang diagnostic push
+        #pragma clang diagnostic ignored "-Wgnu-zero-variadic-macro-arguments"
+    #endif
+    //-----------------------------------------------------------------------------
+    //! CUDA runtime error checking with log and exception, ignoring specific error values
+    #define ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK_IGNORE(cmd, ...)\
+        ::alpaka::uniform_cuda_hip::detail::rtCheckLastError("'" #cmd "' A previous CUDA call (not this one) set the error ", __FILE__, __LINE__);\
```

Comment:
`A previous CUDA call` should probably be `A previous HIP call` ?

Reply:
`A previous API call`?

Reply:
Ah, yes, it is unified, I somehow thought this piece was HIP-only.